expression/builtin_cast.go

```diff
@@ -820,8 +820,13 @@ func (b *builtinCastRealAsTimeSig) evalTime(row chunk.Row) (types.Time, bool, er
 	if isNull || err != nil {
 		return types.Time{}, true, err
 	}
+	// MySQL compatibility: 0 should not be converted to null, see #11203
+	fv := strconv.FormatFloat(val, 'f', -1, 64)
```

Comment:
Should `builtinCastDecimalAsTimeSig` also handle this?

Reply:
`builtinCastDecimalAsTimeSig` handles it via `ParseTimeFromFloatString`